Fix convert error

beninidavide opened the pull request:
In more recent ImageMagick releases, the convert utility generates an error while doing
convert -density 300 example.pdf -resize 25% example.png

Error is
convert: attempt to perform an operation not allowed by the security policy `PDF' @ error/constitute.c/IsCoderAuthorized/449.

Instead of using convert utility from ImageMagick, we can use gs (Ghostscript) to do che conversion

Fixes #9

Werkov commented:
Hi @beninidavide, thanks for correcting this.
Despite the target in the `Makefile` rule, the command is supposed to generate all `example-0.png`, `example-1.png` and `example-2.png` from all the pages of the PDF.
I'm not familiar with Ghostscript but would it be possible to modify the rule so that all pages of the PDF are tranformed into a PNG preview?

beninidavide committed:
Add new features

- Change place for generated files

  Generated files by xelatex go under BUILDDIR (default ./build/) directory
  Generated screenshots by gs go under OUTPUT_IMAGESDIR (default
  ./build/outputimages/)

- Add support for multiple projects

  All projects to compile go in variable PROJECTS

- Change targets

  > make executables
  # produce .pdf

  > make screenshots
  # produce *.png; it depends by executables

  > make
  > make all
  # do everything

beninidavide commented:
Hi @Werkov, as the directory was getting a bit populated with all the files, I did a few changes   
- Change place for generated files
  Generated files by xelatex go under BUILDDIR (default ./build/) directory
  Generated screenshots by gs go under OUTPUT_IMAGESDIR (default ./build/outputimages/)
- Add support for multiple projects
All projects to compile go in variable PROJECTS
- Add targets
 executables -> produce pdf
 screenshots -> produce png
 
 The build directory, compiling 2 projects, will look like
 ```bash
>tree -AFn build
build/
├── anotherExample.aux
├── anotherExample.log
├── anotherExample.nav
├── anotherExample.out
├── anotherExample.pdf
├── anotherExample.snm
├── anotherExample.toc
├── anotherExample.vrb
├── example.aux
├── example.log
├── example.nav
├── example.out
├── example.pdf
├── example.snm
├── example.toc
├── example.vrb
└── outputimages/
    ├── anotherExample-1.png
    ├── anotherExample-2.png
    ├── anotherExample-3.png
    ├── anotherExample-screenshot
    ├── example-1.png
    ├── example-2.png
    ├── example-3.png
    └── example-screenshot
```

Werkov commented:
Thanks.
I've done some integration touches and pushed your commit to main. Hence closing this PR.
